### PR TITLE
[NU-1721] Fix for DatabaseLookupEnricher mixing fields values when it is connected to ignite db 

### PR DESCRIPTION
**Pull request by paw787878**
## Checklist before merge
- [x] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [x] Code is cleaned from temporary changes and commented out lines
- [x] Parts of the code that are not easy to understand are documented in the code
- [x] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [x] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Comment by lukasz-bigorajski** [comment]
I would add a changelog entry describing that issue

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+class DatabaseQueryEnricherPostgresqlTest extends BasePostgresqlQueryEnricherTest with BeforeAndAfterEach {
+
+  import scala.concurrent.duration._
+  import scala.jdk.CollectionConverters._
+
+  override val service =
+    new DatabaseQueryEnricher(postgresqlDbPoolConfig, new MetaDataProviderFactory().create(postgresqlDbPoolConfig))
+
+  override val prepareHsqlDDLs: List[String] = List(
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala:21]: Why is `Hsql` not `sql`?

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    "CREATE TABLE persons (id INT, name VARCHAR(40));",
+    "INSERT INTO persons (id, name) VALUES (1, 'John')"
+  )
+
+  override protected def afterEach(): Unit = {
+    val cleanupStatements = List(
+      "TRUNCATE TABLE persons;",
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala:28]: I guess person or people could be more correct ;)

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    )
+    cleanupStatements.foreach { ddlStr =>
+      val ddlStatement = conn.prepareStatement(ddlStr)
+      try ddlStatement.execute()
+      finally ddlStatement.close()
+    }
+  }
+
+  test("DatabaseQueryEnricherPostgresqlTest#implementation without cache") {
+    val query = "select * from persons where id = ?"
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala:39]: Cannot we extract method to query with enricher?

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      finalState = Some(state)
+    )
+    returnType(service, state).display shouldBe "List[Record{id: Integer, name: String}]"
+    val resultF = implementation.invoke(Context.withInitialId)
+    val result  = Await.result(resultF, 5 seconds).asInstanceOf[java.util.List[TypedMap]].asScala.toList
+    result shouldBe List(
+      TypedMap(Map("name" -> "John", "id" -> 1))
+    )
+
+    conn.prepareStatement("UPDATE persons SET name = 'Alex' WHERE id = 1").execute()
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala:66]: Too many responsibilites for one test. If you want to test update statement you should have separate test for that.

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        Map(
+          DatabaseQueryEnricher.cacheTTLParamName -> null,
+          ParameterName("arg1")                   -> 1
+        )
+      ),
+      dependencies = Nil,
+      finalState = Some(state)
+    )
+    returnType(service, state).display shouldBe "List[Record{id: Integer, name: String}]"
+    val resultF = implementation.invoke(Context.withInitialId)
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala:60]: `resultF` says nothing to me. I would replace that with more meaningful name.

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+    conn.prepareStatement("UPDATE persons SET name = 'Alex' WHERE id = 1").execute()
+    val resultF2 = implementation.invoke(Context.withInitialId.withVariables(Map("arg1" -> 1)))
+    val result2  = Await.result(resultF2, 5 seconds).asInstanceOf[java.util.List[TypedMap]].asScala.toList
+    result2 shouldBe List(
+      TypedMap(Map("name" -> "Alex", "id" -> 1))
+    )
+  }
+
+  test(
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherPostgresqlTest.scala:74]: The same comments as above, also apply to this test too.

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherTest.scala [hunk 7]
```diff
@@ -22,17 +23,65 @@ class DatabaseQueryEnricherTest extends BaseHsqlQueryEnricherTest {
...
+      finalState = Some(state)
+    )
+    returnType(service, state).display shouldBe "List[Record{ID: Integer, NAME: String}]"
+    val resultF = implementation.invoke(Context.withInitialId)
+    val result  = Await.result(resultF, 5 seconds).asInstanceOf[java.util.List[TypedMap]].asScala.toList
+    result shouldBe List(
+      TypedMap(Map("ID" -> 1, "NAME" -> "John"))
+    )
+
+    conn.prepareStatement("UPDATE persons SET name = 'Alex' WHERE id = 1").execute()
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricherTest.scala:66]: Too many reponsiblities for one test.

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BasePostgresqlQueryEnricherTest.scala [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    .withValue("name", ConfigValueFactory.fromAnyRef("db-enricher"))
+    .withValue(
+      "dbPool",
+      ConfigValueFactory.fromMap(
+        postgresqlConfigValues.asJava
+      )
+    )
+
+  override def beforeAll(): Unit = {
+    // TODO_PAWEL is it ok?
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BasePostgresqlQueryEnricherTest.scala]: What about this?

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    "password"        -> password,
+    "url"             -> url
+  )
+
+  def prepareHsqlDDLs: List[String]
+
+  override protected def beforeAll(): Unit = {
+    // DriverManager initializes drivers once per JVM start thus drivers loaded later are skipped.
+    // We must ensue that they are load manually
+    // TODO_PAWEL jaka klasa?
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala]: Is this ok?

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+import com.dimafeng.testcontainers.{ForAllTestContainer, PostgreSQLContainer}
+import org.hsqldb.jdbcDriver
+import org.scalatest.BeforeAndAfterAll
+import org.testcontainers.utility.DockerImageName
+import scala.jdk.CollectionConverters._
+
+import java.sql.{Connection, DriverManager}
+import java.util.UUID
+
+trait WithPostgresqlDB {
```
Comment (lukasz-bigorajski) [inline components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala:12]: Cannot we just move `pl.touk.nussknacker.test.base.db.WithPostgresDbTesting` to the `test-utils` module and reused it in designer and in sql component?

**Review by lukasz-bigorajski — CHANGES_REQUESTED** [review]

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala (hunk 9 above)
Comment (paw787878) [reply components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala]: tak, tutaj została mi klasa z hsql, ostatecznie usunąłem tą linię tutaj i testy działają

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala (hunk 10 above)
Comment (paw787878) [reply components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/WithPostgresqlDB.scala:12]: Rozumiem, że chodzi o połączenie tego treita z traitem WithTestPostgresDb z desginera. Moim zdaniem to nie da za dużo, bo część wspólna ich to jest tylko stworzenie test containera i wyłuskanie  z niego danych do połaczenia się z bazą danych. Czyli dość mało. A też te traity teraz robią jakieś inne rzeczy

File: components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BasePostgresqlQueryEnricherTest.scala (hunk 8 above)
Comment (paw787878) [reply components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BasePostgresqlQueryEnricherTest.scala]: tutaj chyba nie byłem pewien czy też używac lite engine, ale już wiem że to jest ok

**Review by lukasz-bigorajski — APPROVED** [review]

**Pull request merged**